build-system/babel-plugins/babel-plugin-transform-jss/test/fixtures/transform-assertions/should-transform-jss-var/output.mjs

```diff
@@ -21,3 +22,4 @@ const JSS = {
   }
 };
 export const useStyles = () => _classes;
+export const CSS = ".button-21aa4a8{font-size:12px}\n";
```

Comment:
One line above: `export const useStyles = () => _classes;`. Now that I see this code, I'm wondering if it's correct from closure point of view. Is there such a thing as `_classes` map in closure anymore? Or are all `_classes.button` already rewritten to the constant value?

Further, what happens to this code after Closure pass:

```
const classes = useStyles();
className = classes.button;
```

Does it become just:
```
className = "button-21aa4a8"
```

Reply:
Ah. While it is valid, I'm afraid the getter breaks the optimization :(. See [this example](https://closure-compiler.appspot.com/home#code%3D%252F%252F%2520%253D%253DClosureCompiler%253D%253D%250A%252F%252F%2520%2540compilation_level%2520ADVANCED_OPTIMIZATIONS%250A%252F%252F%2520%2540output_file_name%2520default.js%250A%252F%252F%2520%2540formatting%2520pretty_print%250A%252F%252F%2520%2540language_out%2520ECMASCRIPT5_STRICT%250A%252F%252F%2520%253D%253D%252FClosureCompiler%253D%253D%250A%250A%250A%252F**%2520%2540enum%2520%257Bstring%257D%2520*%252F%250Aconst%2520a%2520%253D%2520%257B%2520cx1%253A%2520'hello'%252C%2520cx2%253A%2520'world'%257D%253B%250Aconst%2520getA%2520%253D%2520()%2520%253D%253E%2520a%253B%250A%250Aconsole.log(getA().cx1)%253B%250Aconsole.log(getA().cx2)).

Reply:
Can we just remove the `useStyles()` indirection? https://astexplorer.net/#/gist/1d39dccb98de42b048abba015f2d637a/693d1721957da5b144a40d8873f1c06b6fce637d

Reply:
@jridgewell: there are numerous options for workarounds, but all have their own issues. We currently have `useStyles` call to match the `react-jss` API so that it can be seamlessly swapped out in AMP mode.  Options for us to pursue include:

1. Transform both the `.jss` and its usages. This involves significantly more babel transforming, albeit simple ones.
2. Upstream a Closure fix so that it can optimize this properly
3. Switch libraries from JSS to one that can be optimized with less work.

My favorite options are (2) and (3), but based on where the CSS story is right now, it should be fine to push the effort off for longer.

For now I think its best to merge this as-is even with the deopt. The current JSON.parse solution is fragile and relies on weird assumption being made by Closure that the types of JSS and the parsed value match.

Reply:
Of these three, I think the most realistic and beneficial would be (3). That being said, we (a) need to confirm that an alternative covers our use cases, and (b) we can definitely stay on "unoptimized" JSS for some time if needed.

Reply:
Just wanted to confirm for now that this works. I was always under the impression that Closure simply "threw away" the JS object that represented the enum. If this works, however, it's totally good.

Reply:
Yeah, it works. check out the [appspot example](https://closure-compiler.appspot.com/home#code%3D%252F%252F%2520%253D%253DClosureCompiler%253D%253D%250A%252F%252F%2520%2540compilation_level%2520ADVANCED_OPTIMIZATIONS%250A%252F%252F%2520%2540output_file_name%2520default.js%250A%252F%252F%2520%2540formatting%2520pretty_print%250A%252F%252F%2520%2540language_out%2520ECMASCRIPT5_STRICT%250A%252F%252F%2520%253D%253D%252FClosureCompiler%253D%253D%250A%250A%250A%252F**%2520%2540enum%2520%257Bstring%257D%2520*%252F%250Aconst%2520a%2520%253D%2520%257B%2520%2522cx1%2522%253A%2520'hello'%252C%2520%2522cx2%2522%253A%2520'world'%257D%253B%250Aconst%2520getA%2520%253D%2520()%2520%253D%253E%2520a%253B%250A%250Aconsole.log(getA().cx1)%253B%250Aconsole.log(getA().cx2)). The only issue is that it deopts.


I've also tested via `npx gulp --compiled` and styles seem to work fine